web/web.go

```diff
@@ -582,25 +583,64 @@ func (h *Handler) rules(w http.ResponseWriter, r *http.Request) {
 	h.executeTemplate(w, "rules.html", h.ruleManager)
 }
 
-func (h *Handler) targets(w http.ResponseWriter, r *http.Request) {
-	// Bucket targets by job label
+// Bucket targets by job label
+func bucketTargets(targets []*retrieval.Target) struct {
+	TargetPools map[string][]*retrieval.Target
+} {
 	tps := map[string][]*retrieval.Target{}
-	for _, t := range h.targetManager.Targets() {
+	for _, t := range targets {
 		job := t.Labels().Get(model.JobLabel)
+		if job == "" {
+			job = t.DiscoveredLabels().Get(model.JobLabel)
+		}
 		tps[job] = append(tps[job], t)
 	}
-
 	for _, targets := range tps {
 		sort.Slice(targets, func(i, j int) bool {
 			return targets[i].Labels().Get(labels.InstanceName) < targets[j].Labels().Get(labels.InstanceName)
 		})
 	}
-
-	h.executeTemplate(w, "targets.html", struct {
+	return struct {
 		TargetPools map[string][]*retrieval.Target
 	}{
 		TargetPools: tps,
-	})
+	}
+}
+
+func (h *Handler) serviceDiscovery(w http.ResponseWriter, r *http.Request) {
+	bucketedTargets := bucketTargets(h.targetManager.Targets())
+	bucketedDroppedTargets := bucketTargets(h.targetManager.DroppedTargets())
+	var (
+		targetData = make(map[string][]*retrieval.Target)
+		index      []string
+	)
+	for job, targets := range bucketedTargets.TargetPools {
+		targetData[job] = targets
+	}
+	for job, targets := range bucketedDroppedTargets.TargetPools {
+		if targetData[job] != nil {
+			targetData[job] = append(targetData[job], targets...)
+		} else {
+			targetData[job] = targets
+		}
+	}
+	for job := range targetData {
+		index = append(index, job)
+	}
+	sort.Strings(index)
+	scrapeConfigData := struct {
+		Index   []string
+		Targets map[string][]*retrieval.Target
```

Comment:
A sorted list would be better, Go map iteration order is not guaranteed

Reply:
Would it be acceptable to use the already defined 'Index' list to ensure correct ordering of items in the hashmap when displaying them on the page?